### ISSUE TITLE
Smart lock does not appear

### ISSUE THREAD

**Issue by Jolanrensen** [persandstrom/python-verisure#132]
When retrieving the overview, "doorLockStatusList" appears to be empty. This while we do have a Verisure smart lock.

I haven't checked to open it yet (I don't wanna ring any bells at the security firm for messing with the software too much), but reading the state of the lock would be useful.

The lock does show up in the event log (where I've redacted some IDs):

```json
{
            "deviceArea": "Voordeurslot",
            "deviceType": "DOORLOCK2",
            "eventType": "DOORLOCK_UNLOCK_REMOTE",
            "eventCategory": "UNLOCK",
            "eventId": "REDACTED",
            "eventSource": "GATEWAY",
            "eventTime": "2022-01-19T21:22:03.000Z",
            "userName": "REDACTED",
            "userType": "CUSTOMER",
            "deviceLabel": "REDACTED"
        },
```

Not sure if that helps :)

**Comment by ptz0n** [comment]
Interesting, maybe Verisure have moved the information to another property or endpoint all together?

**Comment by Jolanrensen** [comment]
Could be! Anything I can try to check?

**Comment by ptz0n** [comment]
I don't have any smart lock in my installation. Check the overview:

```bash
vsure user@example.com mypassword overview
```

**Comment by Jolanrensen** [comment]
The doorlock isn't visible in the overview at all. Like I said, the doorLockStatusList is "[]" and for the rest there is no mention of a smart lock. Perhaps they moved it to another endpoint indeed. Are there any docs or other implementations we might be able to look at?

**Comment by persandstrom** [comment]
As you say, it's possible that they have moved the lock info away from the overview. I am not able to verify that since I do not own a lock (yet). There are no docs so I usually set up a man-in-the-middle using Charles proxy (charlesproxy.com) and record the communication between the app and the server. I would be happy to accept a patch with a fix the issue.

**Comment by Jolanrensen** [comment]
Using charlesproxy I see the app connect to https://m-api01.verisure.com

However I seem to be getting 404s when trying to log in with that url

**Comment by persandstrom** [comment]
Oh, that's right. I pointed you in the wrong direction. They have a new api now for the app, and I started a branch for that ( the branch is called "m-api") some time ago. Totally forgot about that. Do not really remember the state, but it is not compatible with home assistant. 

Let me see what I can figure out.

**Comment by Jolanrensen** [comment]
I see, it's a lot different indeed, now using GraphQL

Haha, it appears OpenHAB beat us to it https://github.com/openhab/openhab-addons/blob/da59cdd255a66275dd7ae11dd294fedca4942d30/bundles/org.openhab.binding.verisure/src/main/java/org/openhab/binding/verisure/internal/VerisureBindingConstants.java
I'll actually have a look to see whether OpenHAB can see my doorlock actually!
Edit: Turns out I don't know OpenHAB and simply trying something like this quickly is too steep of a learning curve, haha. In the documentation, they do call it a "Verisure Yaleman SmartLock". I don't know if that's what I have installed in my door as it simply has the Verisure logo, and that's it. It does look earily similar to a danalock v3.

**Comment by persandstrom** [comment]
I have heard from others that the lock status works, so the status does not seem to happen everyone.

Maybe if you could answer a few questions I could try to narrow the issue down in my head:
* Have you ever seen the lock status for this lock in the overview or home-assistant?
* Is it a new lock?
* Which lock model do you have?
* Where are you from? (I guess Netherlands or Belgium :) )
* Are you reading from the correct installation? (I guess you are since you got the event log)

**Comment by Jolanrensen** [comment]
Nope, it does not appear in the overview or Hass, just the Verisure app.
Yes, it's new.
I suspect it's the Danalock v3. At least, that's what it looks like. It just has the Verisure logo on it.
Netherlands :)
Yup. Other sensors work perfectly, also in Hass.

I think we'll need the m-api for the lock. The m-api also seems to support @Home, which provides presence detection.

**Comment by persandstrom** [comment]
Nice, I thought that Yale were the only lock that worked with verisure.

Could you try with the m-api branch of this project and see if it you can find the lock?

**Comment by Jolanrensen** [comment]
After running the login() and set_giid() functions, 
`sesion.request(session.smart_lock())` results in (prettified and redacted):

```json
{
  "data": {
    "installation": {
      "smartLocks": [
        {
          "lockStatus": "LOCKED",
          "doorState": "CLOSED",
          "lockMethod": "REMOTE",
          "eventTime": "2022-01-23T13:45:58.000Z",
          "doorLockType": "DANALOCK3",
          "secureMode": "NOT_APPLICABLE",
          "device": {
            "deviceLabel": "REDACTED",
            "area": "Voordeurslot",
            "__typename": "Device"
          },
          "user": {
            "name": "REDACTED",
            "__typename": "User"
          },
          "__typename": "SmartLock"
        }
      ],
      "__typename": "Installation"
    }
  }
}
```

Let's see if I can open it :) (Edit: LockDoorInput is not yet implemented)

**Comment by Jolanrensen** [comment]
Yes it works :D Needed a slight modification (stolen from OpenHAB):

```python3
@query_func
def door_lock(self,
              deviceLabel: VariableTypes.DeviceLabel,
              code: VariableTypes.Code):
    """Lock door"""
    return {
        "operationName": "DoorLock",
        "variables": {
            "giid": self._giid,
            "deviceLabel": deviceLabel,
            "input": {
                "code": code,
            }
        },
        "query": "mutation DoorLock($giid: String!, $deviceLabel: String!, $input: LockDoorInput!) {\n  DoorLock(giid: $giid, deviceLabel: $deviceLabel, input: $input)\n}\n",  # noqa: E501
    }

@query_func
def door_unlock(self,
                deviceLabel: VariableTypes.DeviceLabel,
                code: VariableTypes.Code):
    """Unlock door"""
    return {
        "operationName": "DoorUnlock",
        "variables": {
            "giid": self._giid,
            "deviceLabel": deviceLabel,
            "input": {
                "code": code,
            },
        },
        "query": "mutation DoorUnlock($giid: String!, $deviceLabel: String!, $input: LockDoorInput!) {\n  DoorUnlock(giid: $giid, deviceLabel: $deviceLabel, input: $input)\n}\n",  # noqa: E501
    }
```

**Comment by niro1987** [comment]
> but it is not compatible with home assistant.

Is there anything we can do to help you out?

**Comment by persandstrom** [comment]
> 

Thanks for asking! 

What is lackning is an integration component in HA which is using the m-api branch. I will not be able to prioritize that myself. If someone would take on that task I could of corse offer some guidance and do any changes needed to the python-verisure lib.

**Comment by niro1987** [comment]
Fixed in #140

**Comment by Jolanrensen** [comment]
If it's fixed, I'll close it, thanks for the awesome work! Unfortunately, I don't have the lock anymore since my door lock was too heavy :( (Thanks, Verisure salesman...)
